src/plugins/mer/mervirtualboxmanager.cpp

```diff
@@ -490,6 +497,27 @@ void MerVirtualBoxManager::shutVirtualMachine(const QString &vmName)
     process->runAsynchronously(arguments);
 }
 
+// accepts void slot(bool ok)
```

Comment:
It is really not clear to you what is the slot used for?

Reply:
It is, but it's not because of this comment. This comment is just noise as it is. Either remove it or modify it so that it is actually useful. I would prefer the latter, as I don't think it should be required that the person who calls this function has to read the code to understand what it does.

Reply:
I didn't mean to ask you if it is clear to you by reading the code. I meant if it is clear to you just from the signature of the function - I assumed that the Callback pattern is familiar to you. I am sure it is.

I used the name "slot", in Qt they use the name "functor" http://doc.qt.io/qt-5/qtimer.html#singleShot-6, in Qt Creator code they usually call it "fi" (they use the QFutureInterface. On other place I suggest we should try to use it as well instead of std::function). No matter of the name used, the pattern is clear to you I believe.

Then what you need to know is what will be passed as arguments to the callback. So it is not that "it just gives name to the parameter". It tells you what is the meaning of the parameter, as "bool" can be used for many things :) I wouldn't do it if it was clear as it is in the case of `void getHostTotalMemorySizeMb(QObject *context, std::function<void(int)> slot)`

Reply:
No, I wouldn't say the parameters "context" and "slot" are clear without any explanation. Sure, the word "slot" does sound a lot like the slots used by Qt signals, but there is no indication that the parameter "context" is somehow associated with the "slot" parameter. And even though I probably could guess what it is, someone else might not. Writing a line or two of comment to explain the parameters and their usage doesn't harm anyone, but it could help someone who isn't familiar with your parameter naming idioms.

Reply:
btw. That Qt documentation link you pasted gives a perfect example how a little documentation clear things up. Imagine that they didn't have that documentation - could you be 100% sure about what "context" and "functor" do, without looking at the source?

Reply:
That's the point that it is not my or our naming idiom - it is well established idiom used by core Qt APIs http://doc.qt.io/qt-5/qobject.html#connect-5 :) just the name "slot" is here used by other existing functions most likely for historical reasons, because with older Qt/C++ it was "QObject *context, const char *slot", and used with the SLOT macro. So unless you are a Qt newbie, you should know very well why the context argument is here.

Reply:
It is established for QObject::connect() sure. But how am I supposed to know that you are using it here? Especially since you are not mentioning anywhere that you are using a callback pattern. 
The next person working on the code might just be Qt newbie - why not give him a hand and write some actually useful comments?

Reply:
Oh, I know quite well that the Qt Creator code is a big undocumented mess. But in my opinion that is not an excuse for us not to be any better.
Personally, I also prefer self-documenting code. The only way I can think of writing this code so that it would not require any comments would be to rename the parameters to e.g. "contextForCallback" and "callback". But that would violate the naming idiom. 
I'm not asking you to comment every little piece of code - just a short explanation that slot is used for a callback and context is tied to that. Rest of the parameters are self-explanatory.
And I don't think you are lazy - you have quite well proved the opposite by spending much longer time arguing this than what it would take to write the comment ;-)

Reply:
You are still concerned about ONE comment. I am concerned about ONE pattern. Following ONE pattern requires to write not just ONE comment, but to write ONE comment on every place where the pattern is applied. Or to simply assume that no ONE of us is a newbie while every ONE experienced Qt (Creator (Mer)) programmer should be able to spot the pattern quickly.

Spot the pattern? ^^^ :)

Reply:
I don't think fixing bad code in one place requires one to fix all the other places at the same time. Even Qt guys don't do that - even though they are using C++11/C++14 features in their NEW code, they haven't changed all their old code to follow the same pattern.
I would be quite happy if we just refused to write bad new code.

Reply:
I think we are mixing consistency with completeness here. I mean if I comment on one place and the newbie looks at other place... Regarding consistency, I think it is enough to be consistent at some sensible scope, like file scope (or cpp/h file pair), so I am ok with gradual adoption of conventions etc. in compliance with this. Ville, thank you for nice discussion, it helps to sort thoughts out in one's head :)